os/arch/arm/src/armv7-m/up_hardfault.c

```diff
@@ -113,10 +115,17 @@ uint32_t system_exception_location;
 
 int up_hardfault(int irq, FAR void *context, FAR void *arg)
 {
+#if defined(CONFIG_DEBUG_HARDFAULT) || !defined(CONFIG_ARMV7M_USEBASEPRI) || defined(CONFIG_BINMGR_RECOVERY)
 	uint32_t *regs = (uint32_t *)context;
-	system_exception_location = regs[REG_R15];
+#endif
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	g_assertpc = regs[REG_R15];
+#endif
 	/* Get the value of the program counter where the fault occurred */
 
+	stm32h745_flash_ecc_handler();
```

Comment:
1. Except this line, all of other changes should be reverted.
2. This file is a common for all of armv7-m architecture. But this is a ST specific function. Let's cover it with the ST chipset config.